lib/cftlib-agent/src/main/java/uk/gov/hmcts/rse/ccd/lib/definitionstore/JsonDefinitionReader.java

```diff
@@ -0,0 +1,175 @@
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.Arrays.asList;
+
+/**
+ * Adds json definition parsing to CCD definition store.
+ * Runs only in definition store.
+ */
+@ConditionalOnClass(CaseDataAPIApplication.class)
+@Component
+public class JsonDefinitionReader extends SpreadsheetParser {
+
+    private static final List<String> FILES = Arrays.stream(SheetName.values())
+            .map(SheetName::getName)
+            .collect(Collectors.toList());
+
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    @Autowired
+    public JsonDefinitionReader(SpreadsheetValidator spreadsheetValidator) {
+        super(spreadsheetValidator);
+    }
+
+    /**
+     * Extend CCD's spreadsheet parser with one that supports JSON.
+     * Definition store expects an xlsx which we extend by supporting
+     * the passing of a file path indicating the directory containing definition json.
+     */
+    @Override
+    public Map<String, DefinitionSheet> parse(InputStream inputStream) throws IOException {
+        var data = inputStream.readAllBytes();
+        var path =  new String(data);
+        try {
+            if (Path.of(path).toFile().exists()) {
+                return fromJson(path);
+            }
+        } catch (InvalidPathException i) {
+            // Treat it as xlsx
+        }
+        return super.parse(new ByteArrayInputStream(data));
+    }
+
+    /**
+     * Avoid an NPE since the list in the base class is initialised by the parse routine.
+     */
+    @Override
+    public List<String> getImportWarnings() {
+        return List.of();
+    }
+
+    @SneakyThrows
+    public static List<Map<String, String>> readPath(String path) {
+        var fi = Paths.get(path).toFile();
+        List<File> files = new ArrayList<>();
+        if (fi.exists()) {
+            files = Files.walk(fi.toPath())
+                    .filter(Files::isRegularFile)
+                    .map(Path::toFile)
+                    .sorted()
+                    .collect(Collectors.toList());
+        }
+        final var file = Paths.get(path + ".json").toFile();
+
+        files.add(file);
+
+        return files.parallelStream()
+                .filter(f -> f.exists() && f.getName().endsWith(".json") && f.canRead())
+                .flatMap(JsonDefinitionReader::readFile)
+                .collect(Collectors.toList());
+    }
+
+    @SneakyThrows
+    private static Stream<Map<String, String>> readFile(File file) {
+        var s = FileUtils.readFileToString(file);
+        // TODO - this is highly inefficient and will be slow if a project uses lots of env vars.
+        for (String s1 : System.getenv().keySet()) {
```

Comment:
it would be slightly more efficient if you created a list of CCD_DEF env vars at the start of the process rather than when processing every file.